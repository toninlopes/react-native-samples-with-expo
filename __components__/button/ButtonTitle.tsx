import { StyleSheet, Text } from "react-native";
import type { ButtonTitleProps } from './types';

const ButtonTitle: React.FC<ButtonTitleProps> = ({ children, ...rest }) => (
  <Text style={styles.text} {...rest}>{children}</Text>
);

const styles = StyleSheet.create({
  text: {
    fontSize: 18,
    alignSelf: 'center',
  },
});

export default ButtonTitle;