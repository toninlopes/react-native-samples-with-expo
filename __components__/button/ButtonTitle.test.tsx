import React from 'react';
import renderer from 'react-test-renderer';

import ButtonTitle from './ButtonTitle';

describe('ButtonTitle', () => {
  test('When there is text.', () => {
    const button = renderer.create(<ButtonTitle>{'Title'}</ButtonTitle>).toJSON();
    expect(button).toMatchSnapshot();
  });
});