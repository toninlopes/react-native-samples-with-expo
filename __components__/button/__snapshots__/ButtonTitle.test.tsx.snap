// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTitle When there is text. 1`] = `
<Text
  style={
    {
      "alignSelf": "center",
      "fontSize": 18,
    }
  }
>
  Title
</Text>
`;
